File: src/main/java/bbarena/model/pitch/BallMove.java
package bbarena.model.pitch;

import java.io.Serializable;

import bbarena.model.team.Player;

public class BallMove implements Serializable {

	private static final long serialVersionUID = 3146629616385694103L;

	public enum BallMoveType {
		SCATTER, PASS, KICK_OFF, THROW_IN, PICK_UP, CATCH, OUT, LOSE;
	}

	private Ball ball = null;

	private BallMoveType moveType = null;

	private Square fromSquare = null;

	private Player fromPlayer = null;

	public BallMove(Ball ball, BallMoveType moveType) {
		this.ball = ball;
		this.fromPlayer = ball.getOwner();
		this.fromSquare = ball.getSquare();
		this.moveType = moveType;
	}

	public boolean isFromPlayer() {
		return this.fromPlayer != null;
	}

	public boolean isFromSquare() {
		return this.fromSquare != null;
	}

	public Player getFromPlayer() {
		return fromPlayer;
	}

	public Square getFromSquare() {
		return fromSquare;
	}

	public Ball getBall() {
		return ball;
	}

	public BallMoveType getMoveType() {
		return moveType;
	}

}

File: src/main/kotlin/bbarena/model/pitch/BallMove.kt
package bbarena.model.pitch

import bbarena.model.team.Player

import java.io.Serializable

data class BallMove(val ball: Ball, val moveType: BallMoveType) : Serializable {

    val fromSquare: Square? = ball.square

    val fromPlayer: Player? = ball.owner

    val isFromPlayer: Boolean
        get() = this.fromPlayer != null

    val isFromSquare: Boolean
        get() = this.fromSquare != null

    enum class BallMoveType {
        SCATTER, PASS, KICK_OFF, THROW_IN, PICK_UP, CATCH, OUT, LOSE
    }

}
